src/NUnitCore/core/Compatibility.cs

```diff
@@ -0,0 +1,298 @@
+﻿using System;
+using System.IO;
+using System.Collections.Generic;
+using System.Reflection;
+
+namespace NUnit.Core
+{
+    public static class Compatibility
+    {
+        private const string WORK_FILE = "COMPATIBILITY.TXT";
+        private const string REPORT_FORMAT = "{0}:{1}:{2}";
+        private const BindingFlags ALL_MEMBERS = 
+            BindingFlags.Instance | BindingFlags.Static | BindingFlags.Public | BindingFlags.NonPublic;
+
+        private static string _workFilePath;
+        private static StreamWriter _writer;
+
+        #region Initialization and Cleanup
+
+        /// <summary>
+        /// This method initializes the working directory and file path
+        /// for compatibility reporting without starting collection.
+        /// It should only be called by the console runner itself in
+        /// the primary AppDomain.
+        /// </summary>
+        public static void Initialize(string workDirectory)
+        {
+            _workFilePath = Path.Combine(workDirectory, WORK_FILE);
+            File.Create(_workFilePath);
+        }
+
+        /// <summary>
+        /// This method is called once in each test AppDomain to
+        /// begin collecting data.
+        /// </summary>
+        /// <param name="workDirectory"></param>
+        public static void BeginCollection(string workDirectory)
+        {
+            _workFilePath = Path.Combine(workDirectory, WORK_FILE);
+
+            _writer = new StreamWriter(_workFilePath, true)
+            {
+                AutoFlush = true
+            };
+        }
+
+        /// <summary>
+        /// This method is called in each test AppDomain after all
+        /// collection is complete.
+        /// </summary>
+        public static void EndCollection()
+        {
+            if (_writer != null)
+            {
+                _writer.Close();
+                _writer = null;
+            }
+        }
+
+        #endregion
+
+        #region Log Issues
+
+        public static TextWriter Writer { get { return _writer; } }
+
+        public static void Error(string location, string message)
+        {
+            Report("Error", location, message);
+        }
+
+        public static void Warning(string location, string message)
+        {
+            Report("Warning", location, message);
+        }
+
+        public static void Note(string location, string message)
+        {
+            Report("Note", location, message);
+        }
+
+        #endregion
+
+        #region Get Issues for Reporting
+
+        public static IEnumerable<Issue> Issues
+        {
+            get
+            {
+                if (_workFilePath == null)
+                    throw new InvalidOperationException("Compatibility work file path not initialized");
+
+                var records = new List<string>();
+
+                StreamReader reader = new StreamReader(_workFilePath);
```

Comment:
Is there any chance of concurrency between writes to this file and reads from this file?